tsl/src/continuous_aggs/create.c

```diff
@@ -1529,27 +1564,46 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 			}
 			else
 			{
-				out->mat_groupcolname_list = lappend(out->mat_groupcolname_list, pstrdup(colname));
+				/*
+				 * Add indexes only for columns that are part of the GROUP BY clause
+				 * and for finals form we skip adding it because we'll not add the
+				 * extra group by columns to the materialization hypertable anymore
+				 */
+				if (!*skip_adding && tle->ressortgroupref > 0)
+					out->mat_groupcolname_list =
+						lappend(out->mat_groupcolname_list, pstrdup(colname));
 			}
+
 			coltype = exprType((Node *) tle->expr);
 			coltypmod = exprTypmod((Node *) tle->expr);
 			colcollation = exprCollation((Node *) tle->expr);
 			col = makeColumnDef(colname, coltype, coltypmod, colcollation);
 			part_te = (TargetEntry *) copyObject(input);
-			/* need to project all the partial entries so that materialization table is filled */
-			part_te->resjunk = false;
+
+			/* keep original resjunk if finalized or not time bucket */
+			if (!finalized || timebkt_chk)
+			{
+				/*
+				 * Need to project all the partial entries so that
+				 * materialization table is filled
+				 */
+				part_te->resjunk = false;
+			}
+
 			part_te->resno = matcolno;
 
 			if (timebkt_chk)
 			{
 				col->is_not_null = true;
 			}
+
 			if (part_te->resname == NULL)
 			{
 				part_te->resname = pstrdup(colname);
 			}
 		}
 		break;
+
 		case T_Var:
 		{
 			PRINT_MATCOLNAME(colbuf, "var", original_query_resno, matcolno);
```

Comment:
I think at this point we should remove this line completely:
```
1612    out->mat_groupcolname_list = lappend(out->mat_groupcolname_list, pstrdup(colname));
```

Reply:
Why? This is part of the old view code, no?

Reply:
This code was introduced between releases, it's not technically part of the old code. Just so that we don't create the index, no other reason.